pkg/glusterfs/Dockerfile

```diff
@@ -10,55 +10,60 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# The base image to use for the final images
+ARG FINAL_BASE=centos
```

Comment:
Does this still build w/ `buildah bud`? I suspect this line and the construct on [L56](https://github.com/gluster/gluster-csi-driver/pull/85/files#diff-1bbd980b5985d3c1ce59f02b8881f15aR56) will break it due to bugs/missing features.

Reply:
Haven't tried. Any instructions on how to do that? Is it just run `buildah bud` from the root of the repo?

Reply:
It's basically a replacement for docker build: `docker build <stuff>` ==> `buildah bud <stuff>`

Reply:
So i should be able to test it by setting `DOCKER_CMD="buildah bud"`?

Reply:
not quite that easy. That would produce `buildah bud build <stuff>`. You need to replace both "docker" and "build" parts of the command line. The other "stuff" should transfer as-is.

Reply:
Okay. Next step, how do I get (the right version of) buildah installed? :)

Reply:
On fedora? ... I just use dnf.
```
$ dnf info buildah
Last metadata expiration check: 0:00:06 ago on Mon 29 Oct 2018 04:08:04 PM EDT.
Installed Packages
Name         : buildah
Version      : 1.4
Release      : 3.git608fa84.fc28
Arch         : x86_64
Size         : 13 M
Source       : buildah-1.4-3.git608fa84.fc28.src.rpm
Repo         : @System
From repo    : updates
Summary      : A command line tool used for creating OCI Images
URL          : https://github.com/containers/buildah
License      : ASL 2.0
Description  : The buildah package provides a command line tool which can be used to
             : * create a working container from scratch
             : or
             : * create a working container from an image as a starting point
             : * mount/umount a working container's root file system for manipulation
             : * save container's root file system layer to create a new image
             : * delete a working container or an image
```

Reply:
Thanks. Given what I had to do for docker, I didn't want to assume sanity for this. ;)

Reply:
It looks like your original change already broke buildah, as it doesn't recognize `--target`. Should I try refactoring all this so that buildah runs?

Reply:
Yeah... we need to keep buildah working. Of course I'm not sure how we're going to get the coverage data out w/o being able to refer to the intermediate image or splitting into multiple dockerfiles. And having multiple dockerfiles will prevent us from autobuilding on docker hub.
For now, I suppose pulling it forward into the final image then copying it out is good enough... it's small.

BTW, a quick search turned up the bug in buildah: containers/buildah#632

Reply:
Now I'm running into a `no build container available` error that I can't find any diagnosis for online. Ideas?

Reply:
Okay, I managed to figure out the initial error... need to specify the registry `docker.io`. But now I'm running into the broader issue that buildah doesn't seme to support multi-stage builds at all. Thus, if we want to even keep your recent changes we are inherently breaking buildah support. Should I modify this PR to move away from multi-stage builds, or can it move ahead as-is?

Reply:
@jarrpa is above issue resolved at ur end ?

Reply:
Ah, yeah. Had to add `docker.io` to the base image names.